cscs-checks/libraries/io/hdf5_compile_run.py

```diff
@@ -74,3 +75,15 @@ def __init__(self, lang, linkage):
 
         self.maintainers = ['SO', 'RS']
         self.tags = {'production', 'craype'}
+
+    @rfm.run_before('compile')
+    def cray_linker_workaround(self):
+        # FIXME: static compilation yields a link error in case of
+        # PrgEnv-cray(Cray Bug #255707)
+        if not (self.linkage == 'static' and
+                self.current_system.name == 'dom' and
+                self.current_environ.name == 'PrgEnv-cray'):
+            return
+
+        self.variables = {'LINKER_X86_64': '/usr/bin/ld',
```

Comment:
@teojgo, I think it is easier on the Cray to define
`{'ALT_LINKER': '/usr/bin/ld'}`. 
This way you do not have to set the architecture specific ones.

Reply:
@teojgo If that works, I agree with @victorusu

Reply:
@victorusu, @vkarak it seems to work fine, thanks.